.golangci.yaml

```
@@ -1,36 +1,112 @@
+version: "2"
 run:
   concurrency: 4
-  deadline: 10m
-
 linters:
-  disable:
-  - unused
   enable:
-  - revive
-
-skip-files:
-- "zz_generated\\..*\\.go$"
-
-issues:
-  exclude-use-default: false
-  exclude:
-  # revive
-  - var-naming # ((var|const|struct field|func) .* should be .*
-  - dot-imports # should not use dot imports
-  - package-comments # package comment should be of the form
-  - indent-error-flow # if block ends with a return statement, so drop this else and outdent its block
-  - unexported-return # exported func .* returns unexported type .*, which can be annoying to use
-  - "exported: (type|func) name will be used as .* by other packages, and that stutters;"
-  # typecheck:
-  - "undeclared name: `.*`"
-  - "\".*\" imported but not used"
-  exclude-rules:
-    - linters:
-        - staticcheck
-      text: "SA1019:" # Excludes messages where deprecated variables are used
-
-linters-settings:
-  revive:
+    - ginkgolinter
+    - importas
+    - misspell
+    - nilerr
+    - revive
+    - unconvert
+    - unparam
+    - whitespace
+  settings:
+    importas:
+      alias:
+        # External imported packages
+        - pkg: k8s.io/api/(\w+)/(v[\w\d]+)
+          alias: $1$2
+        - pkg: k8s.io/apimachinery/pkg/apis/(\w+)/(v[\w\d]+)
+          alias: $1$2
+        - pkg: k8s.io/apimachinery/pkg/api/([^m]\w+)
+          alias: api${1}
+        - pkg: k8s.io/apimachinery/pkg/util/(\w+)
+          alias: util${1}
+        - pkg: k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1
+          alias: vpaautoscalingv1
+        - pkg: sigs.k8s.io/controller-runtime/pkg/client/fake
+          alias: fakeclient
+        - pkg: sigs.k8s.io/controller-runtime/pkg/log/zap
+          alias: logzap
+        - pkg: sigs.k8s.io/controller-runtime/pkg/log
+          alias: logf
+        - pkg: go.uber.org/mock/gomock
+          alias: gmock
+        # gardener/gardener packages
+        - pkg: github.com/gardener/gardener/pkg/component/(\w+)/constants
+          alias: ${1}constants
+        - pkg: github.com/gardener/gardener/extensions/pkg/webhook
+          alias: extensionswebhook
+        - pkg: github.com/gardener/gardener/extensions/pkg/util/secret/manager
+          alias: extensionssecretsmanager
```

```suggestion
          alias: extensionssecretmanager
```

If it is already used with the additional `s`, just ignore